alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -95,6 +100,22 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
         return clusterState.routingTable.hasIndex(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
     }
 
+    /**
+     * From given index mapping node, extracts fieldName -> fieldProperties pair
+     */
+    fun extractField(node: MutableMap<String, Any>, currentPath: String): Pair<String, MutableMap<String, Any>> {
+        if (node.containsKey(PROPERTIES)) {
+            return extractField(node.get(PROPERTIES) as MutableMap<String, Any>, currentPath)
+        } else if (node.containsKey(NESTED)) {
+            return extractField(node.get(NESTED) as MutableMap<String, Any>, currentPath)
```

Comment:
Is it possible for a node to contain `PROPERTIES` and `NESTED`?

Reply:
I don't think so. They should be "lonely" interim nodes

Reply:
Yea taking more of a look into that, seems to be right